specs/opcode/94MCOPY.md

```diff
@@ -0,0 +1,40 @@
+# MCOPY opcode
+
+## Procedure
+
+The `MCOPY` opcode pops `dest_offset`, `offset` and `size` from the stack.
+It then copies `size` bytes in the current environment from memory at an `offset` to the memory at a `dest_offset`. For out-of-bounds scenarios - the memory is extended with respective gas cost applied.
+
+The gas cost of `MCOPY` opcode consists of two parts:
+
+1. A constant gas cost: `3 gas`
+2. A dynamic gas cost: cost of memory expansion and copying (variable depending on the `size` copied to memory)
+
+## Constraints
+
+1. OpcodeId check
+
+- opId == OpcodeId(0x5E)
+
+2. State Transition
+
+- rw_counter += 3 stack_reads + ???
+- stack_pointer += 3
```

Comment:
I think it should be 3 here, at least because it's 3 in `RETURNDATACOPY` or in `CALLDATACOPY`.
But in the (WIP) scroll impl of MCOPY it's 4:
https://github.com/scroll-tech/zkevm-circuits/pull/1209/files#diff-8f22d99f71cb4ed2a7ed91d2d25ace314f20d903a8e32af87036b12c4d9f4b27R94

Reply:
I see 3 on scroll too though?

`            stack_pointer: Transition::Delta(3.expr()),`

Reply:
It was 4 before, probably a typo; Thanks!